Lib/typing.py

```diff
@@ -516,6 +521,22 @@ def TypeAlias(self, parameters):
     raise TypeError(f"{self} is not subscriptable")
 
 
+@_SpecialForm
+def Concatenate(self, parameters):
+    """Used in conjunction with ParamSpec and Callable to represent a higher
+    order function which add, removes or transform parameters of a Callable.
```

Comment:
Maybe add a reference to PEP 612 and an example, e.g. `Callable[Concatenate[int, P], int]`? (No need to explain `P`.)